docs/customization/controller.rst

```diff
@@ -0,0 +1,167 @@
+                        $this->metadata->getName() => $product,
+                    ])
+                ;
+            }
+
+            return $this->viewHandler->handle($configuration, $view);
+        }
+    }
+
+2. In order to use your controller and its actions you need to configure it in the ``app/config/config.yml``.
+
+.. code-block:: yaml
+
+    sylius_product:
+        resources:
+            product:
+                classes:
+                    controller: AppBundle\Controller\ProductController
+
+How to customize a Standard Controller?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would like to send some kind of emails (which are not resources) after something has been purchased in your shop - to do this you should modify an ``afterPurchaseAction`` on the ``OrderController``.
+
+1. Create a new Controller class under the ``AppBundle/Controller/Frontend`` namespace.
+
+Run ``$ php app/console debug:container sylius.controller.frontend.order``.
+
+Your class needs to be extending this base class.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Controller\Frontend;
+
+    use Sylius\Bundle\WebBundle\Controller\Frontend\Account\OrderController as BaseOrderController;
+    use Sylius\Bundle\PayumBundle\Request\GetStatus;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+
+    class OrderController extends BaseOrderController
+    {
+        /**
+         * @param Request $request
+         *
+         * @return Response
+         */
+        public function afterPurchaseAction(Request $request)
+        {
+            $token = $this->getHttpRequestVerifier()->verify($request);
+            $this->getHttpRequestVerifier()->invalidate($token);
+
+            $status = new GetStatus($token);
+            $this->getPayum()->getGateway($token->getGatewayName())->execute($status);
+            $payment = $status->getFirstModel();
+            $order = $payment->getOrder();
+            $this->checkAccessToOrder($order);
+
+            $orderStateResolver = $this->get('sylius.order_processing.state_resolver');
+            $orderStateResolver->resolvePaymentState($order);
+            $orderStateResolver->resolveShippingState($order);
+
+            $this->getOrderManager()->flush();
+
+            //
+            $emailManager = $this->get('sylius.email_manager.order');
+            $emailManager->sendConfirmationEmail($order);
+
+            return $this->redirectToRoute('sylius_checkout_thank_you');
+        }
+    }
+
+2. The next thing you have to do is to override the ``sylius.controller.frontend.order.class`` parameter in ``AppBundle/Resources/config/services.yml``.
+
+.. code-block:: yaml
+
+    parameters:
+        sylius.controller.frontend.order.class: AppBundle\Controller\Frontend\OrderController
+
+From now on your ``afterPurchaseAction`` of the ``OrderController`` will be sending emails.
```

Comment:
``From now, "afterPurchaseAction" of the "OrderController" will also send emails in addition to its default behaviour.``? Sth like this? This sentence just looks strange for me 😄